server/manager/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -884,8 +886,9 @@ private long balanceTablets() {
       gatherTableInformation(Set<TServerInstance> currentServers) {
     final long rpcTimeout = getConfiguration().getTimeInMillis(Property.GENERAL_RPC_TIMEOUT);
     int threads = getConfiguration().getCount(Property.MASTER_STATUS_THREAD_POOL_SIZE);
-    ExecutorService tp =
-        threads == 0 ? Executors.newCachedThreadPool() : Executors.newFixedThreadPool(threads);
+    ExecutorService tp = threads == 0
+        ? Executors.newCachedThreadPool(new NamingThreadFactory("GatherTableInformation"))
+        : Executors.newFixedThreadPool(threads, new NamingThreadFactory("GatherTableInformation"));
```

Comment:
This is probably a little easier to read if you assign the ThreadFactory to a variable first, like the following (formatting probably isn't right):

```suggestion
    var threadFactory = new NamingThreadFactory("GatherTableInformation");
    ExecutorService tp = threads == 0
        ? Executors.newCachedThreadPool(threadFactory)
        : Executors.newFixedThreadPool(threads, threadFactory);
```